### PR TITLE
add comments, link to docs in defineAuth template

### PR DESCRIPTION
**Pull request by josefaidt**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import { defineAuth } from '@aws-amplify/backend';
+import { defineAuth } from '@aws-amplify/backend'
 
+/** @see https://docs.amplify.aws/gen2/build-a-backend/auth */
 export const auth = defineAuth({
   loginWith: {
     email: true,
   },
-});
+})
```
Comment (josefaidt) [inline packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts:8]:
```suggestion
});
```

File: packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
-import { defineAuth } from '@aws-amplify/backend';
+import { defineAuth } from '@aws-amplify/backend'
```
Comment (josefaidt) [inline packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts:1]:
```suggestion
import { defineAuth } from '@aws-amplify/backend';
```

File: packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts [hunk 3]
```diff
@@ -1,7 +1,28 @@
...
+ * Define and configure your auth resource
+ * When used alongside data, it is automatically configured as an auth provider for data
+ * @see https://docs.amplify.aws/gen2/build-a-backend/auth
+ */
 export const auth = defineAuth({
   loginWith: {
     email: true,
+    // add social providers
+    externalProviders: {
+      // first, create your secrets using `amplify sandbox secret`
```
Comment (swaminator) [inline packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts:13]: @see: https://docs.amplify.aws/gen2/deploy-and-host/sandbox-environments/features/#setting-secrets

File: packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts [hunk 4]
```diff
@@ -1,7 +1,28 @@
...
+      // then, import `secret` from `@aws-amplify/backend`
+      // loginWithAmazon: {
+      //   clientId: secret('LOGINWITHAMAZON_CLIENT_ID'),
+      //   clientSecret: secret('LOGINWITHAMAZON_CLIENT_SECRET'),
+      // }
+    },
```
Comment (swaminator) [inline packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts:19]: I'd also add MFA as when enabled with SMS, it requires the same prerequisites for sending SMS messages via Amazon SNS.

**Review by swaminator — CHANGES_REQUESTED** [review]

File: packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts (hunk 3 above)
Comment (josefaidt) [reply packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts]: added

File: packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts (hunk 4 above)
Comment (josefaidt) [reply packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts:22]: added

**Comment by swaminator** [comment]
Ship it!

**Review by swaminator — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Pull request merged**